program.go

```diff
@@ -108,16 +97,15 @@ type compiledProgram struct {
 }
 
 //nolint:nilnil // non-existence of violations is intentional
-func (expr compiledProgram) eval(bindings *variable, cfg *validationConfig) (*Violation, error) {
-	now := globalNowPool.Get(cfg.nowFn)
-	defer globalNowPool.Put(now)
-	bindings.Next = &variable{
-		Next: now,
-		Name: "rules",
-		Val:  expr.Rules,
+func (expr compiledProgram) eval(activation *variables, cfg *validationConfig) (*Violation, error) {
+	activation.NowFn = cfg.nowFn
+	if expr.Rules != nil {
+		activation.Rules = expr.Rules.Interface()
 	}
```

Comment:
Isn't the value of `rules` the same for all programs in a program set? Why are we passing it around with each programs?